### PR TITLE
Blaze Manage Campaigns: Present campaign details view from campaigns list

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Part of #20765 

## Description
This PR adds an API to `BlazeFlowCoordinator` for presenting the details web view for a campaign. The existing Blaze web views logic was refactored and reused. The PR also hooks up the new API to the campaigns list.

The following are still missing:
- Analytics tracking
- Presenting the details view from the dashboard card
- Use remotely configured params for the web view URLs instead of constants (Ref: #20766)

## Testing Instructions

Before testing, replace `BlazeCampaignsViewController.fetchCampaigns()` with the following:
```swift
private func fetchCampaigns() {
    isLoading = true

    BlazeService.createService()?.getRecentCampaigns(for: blog, completion: { [weak self] (result) in
        self?.isLoading = false
        switch result {
        case .success(let success):
            self?.campaigns = success.campaigns ?? []
        case .failure(let failure):
            print("Failed to fetch recent blaze campaigns: \(failure.localizedDescription)")
            break
        }
    })
}
```

1. Run the Jetpack app
2. Log in using the test account mentioned here: p1687419588093579/1687363386.825549-slack-C04LJFS1G5P 
3. Tap on the Blaze dashboard card header
4. The campaigns list should be displayed and populated
5. Tap on any campaign
6. Make sure the campaign details are presented in a web view ✅ 

## Regression Notes
1. Potential unintended areas of impact
The original blaze flow for creating a campaign.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Relied partially on `BlazeCreateCampaignWebViewModelTests` but also tested manually.

3. What automated tests I added (or what prevented me from doing so)
`BlazeCampaignDetailsWebViewModelTests`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [x] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blaze/Webview/BlazeFlowCoordinator.swift [hunk 1]
```diff
@@ -108,4 +113,27 @@ import UIKit
...
+        let blazeViewController = BlazeWebViewController(delegate: nil)
+        let viewModel = BlazeCampaignDetailsWebViewModel(source: source,
+                                                         blog: blog,
+                                                         campaignID: campaignID,
+                                                         view: blazeViewController)
+        blazeViewController.viewModel = viewModel
+        let navigationViewController = UINavigationController(rootViewController: blazeViewController)
+        navigationViewController.overrideUserInterfaceStyle = .light
+        navigationViewController.modalPresentationStyle = .formSheet
+        viewController.present(navigationViewController, animated: true)
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Blaze/Webview/BlazeFlowCoordinator.swift:137]: I suggest pushing it in the navigation stack:

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/e931d245-2fc2-4215-b3fb-178b0f6af4b2

The "Done" button will need to go.

File: WordPress/Classes/ViewRelated/Blaze/Webview/BlazeCampaignDetailsWebViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import Foundation
+
+class BlazeCampaignDetailsWebViewModel: BlazeWebViewModel {
+
+    // MARK: Private Variables
+
+    private let source: BlazeSource
+    private let blog: Blog
+    private let campaignID: Int
+    private let view: BlazeWebView
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Blaze/Webview/BlazeCampaignDetailsWebViewModel.swift]: There may be a retain cycle here.

Ideally, ViewModel should not know about the View.

**Comment by kean** [comment]
@hassaanelgarem the details page doesn't support Dark Mode and Dynamic Type. I wonder if there is anything we can do about it 🤔

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/2daf814e-08b6-4b19-bb7e-661f9cdfe783

**Comment by hassaanelgarem** [comment]
> The details page doesn't support Dark Mode and Dynamic Type. I wonder if there is anything we can do about it 🤔

@kean We've discussed the dark mode issue when implementing the campaign creation flow, but after some investigation, we couldn't find an easy way to do it without adding too much scope. Do you think it's feasible to implement?

File: WordPress/Classes/ViewRelated/Blaze/Webview/BlazeFlowCoordinator.swift (hunk 1 above)
Comment (hassaanelgarem) [reply WordPress/Classes/ViewRelated/Blaze/Webview/BlazeFlowCoordinator.swift:137]: In the Figma design (Ref: 1Osrn9gb2xLaKG6lIft3WC-fi-1461_9777), it's presented in a modal. But yes, that's a good idea, especially cause of the chevron. 

@osullivanchris What do you think?

File: WordPress/Classes/ViewRelated/Blaze/Webview/BlazeCampaignDetailsWebViewModel.swift (hunk 2 above)
Comment (hassaanelgarem) [reply WordPress/Classes/ViewRelated/Blaze/Webview/BlazeCampaignDetailsWebViewModel.swift]: > There may be a retain cycle here.

Good catch. Addressed in 1cf2cb1cc9fcf14645e84244ef6f34be950653af 👍 

> Ideally, ViewModel should not know about the View.

Good point, but that might require a big refactor that's out of the scope of this PR, IMHO.

File: WordPress/Classes/ViewRelated/Blaze/Webview/BlazeFlowCoordinator.swift (hunk 1 above)
Comment (osullivanchris) [reply WordPress/Classes/ViewRelated/Blaze/Webview/BlazeFlowCoordinator.swift:137]: @hassaanelgarem @kean it feels smooth in the back stack. The only advantage I see of the modal is that its a bit clearer that you're going out of the app to a webview. In cases like preview or go to wp-admin, that seems like a good thing. But here we probably want to de-emphasise the fact its a web view if we're doing it, so we can go with this if you both think its better.

BTW I assume 'back' and 'Done' do the same thing here in the nav bar.

File: WordPress/Classes/ViewRelated/Blaze/Webview/BlazeCampaignDetailsWebViewModel.swift (hunk 2 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Blaze/Webview/BlazeCampaignDetailsWebViewModel.swift]: I agree. WebViews also don't really play well with ViewModels, so I think it's OK.

File: WordPress/Classes/ViewRelated/Blaze/Webview/BlazeFlowCoordinator.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Blaze/Webview/BlazeFlowCoordinator.swift:137]: > The only advantage I see of the modal is that its a bit clearer that you're going out of the app to a webview

Great point, especially given the design differences and lack of dark mode support on the web. I opened a thread to discuss it: p1687806482680829-slack-C04LJFS1G5P.

**Comment by kean** [comment]
@hassaanelgarem the details page is relatively simple, yes.

**Review by kean — APPROVED** [review]

**Pull request merged**